kore/src/Kore/Simplify/Predicate.hs

```diff
@@ -325,22 +328,51 @@ normalizeTop _ = pure (MultiOr.singleton MultiAnd.top)
  external solver or for the user, and the un-expanded form is more compact.
 -}
 simplifyNot ::
-    forall simplifier sort.
+    forall simplifier.
     Monad simplifier =>
-    Not sort NormalForm ->
+    MonadSimplify simplifier =>
+    SideCondition RewritingVariableName ->
+    Not () NormalForm ->
     simplifier NormalForm
-simplifyNot Not{notChild = multiOr, notSort} = do
-    disjunctiveNormalForms <- Logic.observeAllT $ do
-        multiAnd <- Logic.scatter multiOr
-        normalizeNotAnd Not{notSort, notChild = multiAnd} & lift
-    normalizeMultiAnd (MultiAnd.make disjunctiveNormalForms)
+simplifyNot sideCondition Not{notChild = multiOr} = do
+    -- try user-defined rules first
+    mbResult <- runMaybeT applyUserDefined
+    maybe standardSimplification pure mbResult
```

Comment:
I'm actually not so sure about this, let me explain:

The `standardSimplification` basically pushes the `#Not` inside the dnf, and redistributes stuff so that the result is back in dnf, so you get `((#Not p1) #And ... #And (#Not pn)) #Or ...`. It might make more sense to try to apply any user-defined simplification rules to this form, because it's probably what the users are going to see. I don't think they'll ever see the `#Not (conj1 #Or ... #Or conjn)` form.

Reply:
Sure, we can swap the order and always do the standard simplification